src/AbstractPluginManager.php

```diff
@@ -58,17 +58,6 @@ abstract class AbstractPluginManager extends ServiceManager implements PluginMan
      */
     public function __construct($configInstanceOrParentLocator = null, array $config = [])
     {
-        if (
-            $configInstanceOrParentLocator instanceof PsrContainerInterface
-            && ! $configInstanceOrParentLocator instanceof ContainerInterface
-        ) {
-            /**
-             * {@see \Laminas\ServiceManager\Factory\FactoryInterface} typehints
-             * against interop container and as such cannot accept non-interop
-             * psr container. Decorate it as interop.
-             */
-            $configInstanceOrParentLocator = new PsrContainerDecorator($configInstanceOrParentLocator);
-        }
         if (
             null !== $configInstanceOrParentLocator
```

Comment:
Psalm confused here? Or did we really mess up the `$configInstanceOrParentLocator` type? :thinking:

Reply:
Not sure if I can follow but yes, its already a Union type.

Reply:
Ah, I've suppressed that for now as I dont really know how to fix that.
I think that this was an error in the baseline and now popped up due to the fact that we removed the Interop annotation.